apps/dashboard/lib/trpc/routers/deploy/project/list.ts

```diff
@@ -1,30 +1,68 @@
 import { ratelimit, requireUser, requireWorkspace, t, withRatelimit } from "@/lib/trpc/trpc";
-import { TRPCError } from "@trpc/server";
+import { deployments, domains, projects } from "@unkey/db/src/schema";
+
+type ProjectRow = {
+  id: string;
+  name: string;
+  slug: string;
+  updated_at: number | null;
+  git_repository_url: string | null;
+  live_deployment_id: string | null;
+  git_commit_message: string | null;
+  git_branch: string | null;
+  git_commit_author_name: string | null;
+  git_commit_timestamp: number | null;
+  runtime_config: Deployment["runtimeConfig"] | null;
+  domain: string | null;
+};
 
 export const listProjects = t.procedure
   .use(requireUser)
   .use(requireWorkspace)
   .use(withRatelimit(ratelimit.read))
   .query(async ({ ctx }) => {
-    return await db.query.projects
-      .findMany({
-        where: (table, { eq }) => eq(table.workspaceId, ctx.workspace.id),
-        columns: {
-          id: true,
-          name: true,
-          slug: true,
-          updatedAt: true,
-          gitRepositoryUrl: true,
-          liveDeploymentId: true,
-        },
-      })
-      .catch((error) => {
-        console.error("Error querying projects:", error);
-        throw new TRPCError({
-          code: "INTERNAL_SERVER_ERROR",
-          message:
-            "Failed to retrieve projects due to an error. If this issue persists, please contact support.",
-        });
-      });
+    const result = await db.execute(sql`
+      SELECT
+        ${projects.id},
+        ${projects.name},
+        ${projects.slug},
+        ${projects.updatedAt},
+        ${projects.gitRepositoryUrl},
+        ${projects.liveDeploymentId},
+        ${deployments.gitCommitMessage},
+        ${deployments.gitBranch},
+        ${deployments.gitCommitAuthorName},
+        ${deployments.gitCommitTimestamp},
+        ${deployments.runtimeConfig},
+        ${domains.domain}
+      FROM ${projects}
+      LEFT JOIN ${deployments}
+        ON ${projects.liveDeploymentId} = ${deployments.id}
+        AND ${deployments.workspaceId} = ${ctx.workspace.id}
+      LEFT JOIN ${domains}
+        ON ${projects.id} = ${domains.projectId}
+        AND ${domains.workspaceId} = ${ctx.workspace.id}
+      WHERE ${projects.workspaceId} = ${ctx.workspace.id}
+      ORDER BY ${projects.updatedAt} DESC
+    `);
+
+    return (result.rows as ProjectRow[]).map(
+      (row): Project => ({
+        id: row.id,
+        name: row.name,
+        slug: row.slug,
+        updatedAt: row.updated_at,
+        gitRepositoryUrl: row.git_repository_url,
+        liveDeploymentId: row.live_deployment_id,
+        commitTitle: row.git_commit_message ?? "[DUMMY] Initial commit",
+        branch: row.git_branch ?? "main",
+        author: row.git_commit_author_name ?? "[DUMMY] Unknown Author",
+        commitTimestamp: row.git_commit_timestamp ?? Date.now() - 86400000,
+        regions: row.runtime_config?.regions?.map((r) => r.region) ?? ["us-east-1"],
```

Comment:
this is bad, the default should absolutely not be some random region. The control plane already create a proper `runtimeConfig`

Reply:
That `region` and other `[DUMMY ]` data is just for nicer UI so we can find the stylistic issues on the frontend. Everything starts with `[DUMMY ]` or hardcoded stuff will be removed before demo.